pkg/node/compute.go

```diff
@@ -290,3 +249,80 @@ func NewComputeNode(
 func (c *Compute) Cleanup(ctx context.Context) {
 	c.cleanupFunc(ctx)
 }
+
+func NewBidder(
+	config ComputeConfig,
+	publishers publisher.PublisherProvider,
+	storages storage.StorageProvider,
+	executors executor.ExecutorProvider,
+	runningCapacityTracker capacity.Tracker,
+	enqueuedCapacityTracker capacity.Tracker,
+	nodeID string,
+	executionStore store.ExecutionStore,
+	computeCallback compute.Callback,
+	bufferRunner *compute.ExecutorBuffer,
+	apiServer *publicapi.Server,
+	calculator capacity.UsageCalculator,
```

Comment:
When should we use Params and when should we just inline all the parameters.

Reply:
Yeah, this brings up a good point. We should probably give our usage of the `*Params` pattern more thought. It makes it easy to forget to pass values to a constructor when things change unless we inline them as you've said e.g
```go
// can panic when a new field is added to params but not populated in the constructor
svr := NewServer(ServerParams{
	Host: "0.0.0.0",
	Port: 80,
	Config: []byte("whatever"),
})
// will fail to compile if a new field is added to params
svr := NewServer(ServerParams{
	"0.0.0.0",
	80,
	[]byte("whatever"),
})
```
I believe it is considered a best practice to provide field names when initializing a structure, but the catch is we lose compile time guarantees when they change, I am trying to avoid that here, but don't think this is any better. I'll revert it. 

I think the initial intent of the `*Params` pattern was to make it easier to adopt a dependency injection framework (do I remember this correctly @wdbaruni?). For example, if we use uber.fx `*Params` are easy to migrate to: https://uber-go.github.io/fx/parameter-objects.html#using-parameter-objects

Reply:
I do think this is okay for now, really just highlighting that it might be a good candidate for a future PR